### PR TITLE
Add conditional help link

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?
Adds new conditional header `Help` service link based on conditions for business survey described [here](https://trello.com/c/fzjktoqK).

### How to review 
Tests were already in place for this change so it was only a question of tweaking existing ones.

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by petechd** [comment]
[Conditional links: Help (2 SP)](https://trello.com/c/fzjktoqK/5304-conditional-links-help-2-sp)

File: app/survey_config/business_config.py [hunk 1]
```diff
@@ -63,33 +63,40 @@ def _get_account_service_help_url(
         return f"{self.base_url}/help"
 
     def get_service_links(
-        self, sign_out_url: str, *, is_authenticated: bool, ru_ref: Optional[str]
+        self,
+        sign_out_url: str,
+        *,
+        is_authenticated: bool,
+        theme: bool,
```
Comment (MebinAbraham) [inline app/survey_config/business_config.py:70]: The variable should be named for a bool. I expected `theme` to be a string. Maybe something like `cookie_has_theme` etc.

File: app/survey_config/business_config.py [hunk 2]
```diff
@@ -63,33 +63,40 @@ def _get_account_service_help_url(
...
+                            self.account_service_my_account_url,
+                            id="header-link-my-account",
+                        ).__dict__,
+                        HeaderLink(
+                            lazy_gettext("Sign out"),
+                            sign_out_url,
+                            id="header-link-sign-out",
+                        ).__dict__,
+                    ]
+                )
```
Comment (MebinAbraham) [inline app/survey_config/business_config.py:99]: The indentation can be avoided by using a ternary operator. `is_authenticated` doesn't need to be nested.

**Review by MebinAbraham — COMMENTED** [review]
Minor issues.

File: app/survey_config/business_config.py (hunk 1 above)
Comment (petechd) [reply app/survey_config/business_config.py:70]: Added, thanks!

**Review by berroar — APPROVED** [review]
Tested, looks good to me 👍

File: tests/integration/test_header_links.py [hunk 3]
```diff
@@ -86,7 +86,7 @@ def test_links_not_in_header_when_no_session(self):
         # Then
         self.assert_my_account_link_does_not_exist()
         self.assert_sign_out_link_does_not_exist()
-        self.assert_help_link_exist_not_authenticated()
+        self.assert_help_link_does_not_exist()
```
Comment (MebinAbraham) [inline tests/integration/test_header_links.py:89]: At the moment, we have test for session vs no session. We should have one for no session but still got the cookie.

Comment (petechd) [reply tests/integration/test_header_links.py:108]: Added a test where session is gone but cookie still exists, thanks!

Comment (petechd) [reply tests/integration/test_header_links.py:127]: And added one now for social theme as well.

File: tests/integration/test_header_links.py [hunk 4]
```diff
@@ -76,6 +76,44 @@ def test_links_in_header_when_valid_session(self):
...
+
+        # When
+        self.assertStatusCode(302)
+        self.get("questionnaire/")
+
+        # Then
+        self.assertInUrl("questionnaire/")
+        cookie = self.getCookie()
+        self.assertIsNotNone(cookie.get("theme"))
+        self.assertEqual(cookie.get("theme"), "default")
```
Comment (MebinAbraham) [inline tests/integration/test_header_links.py:93]: The first assert is covered by the second, so the `self.assertIsNotNone` can be removed.

File: tests/integration/test_header_links.py [hunk 5]
```diff
@@ -76,6 +76,44 @@ def test_links_in_header_when_valid_session(self):
...
+
+        # When
+        self.assertStatusCode(302)
+        self.get("questionnaire/")
+
+        # Then
+        self.assertInUrl("questionnaire/")
+        cookie = self.getCookie()
+        self.assertIsNotNone(cookie.get("theme"))
+        self.assertEqual(cookie.get("theme"), "social")
```
Comment (MebinAbraham) [inline tests/integration/test_header_links.py:112]: Same as above.

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**